api/v1/storagecluster_types.go

```diff
@@ -447,6 +447,19 @@ type EncryptionSpec struct {
 	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$
 	StorageClassName     string                   `json:"storageClassName,omitempty"`
 	KeyManagementService KeyManagementServiceSpec `json:"kms,omitempty"`
+	// KeyRotation defines options for Key Rotation.
+	// +optional
+	KeyRotation KeyRotationSpec `json:"keyRotation,omitempty"`
+}
+
+// KeyRotationSpec represents the settings for Key Rotation.
+type KeyRotationSpec struct {
+	// Enabled represents whether the key rotation is enabled.
+	// +optional
+	Enabled bool `json:"enabled,omitempty"`
+	// Schedule represents the cron schedule for key rotation.
+	// +optional
+	Schedule string `json:"schedule,omitempty"`
```

Comment:
`omtiempty` is only helpful when the object is a pointer. We can remove it.

Let's add default to Schedule and keep it as @weekly using kubebuilder annotation too.

Reply:
I did not add default values on purpose, does not make sense to add it even if cluster-wide encryption is not opted by the user...

Reply:
instead let `GetKeyRotationSpec` handle it at runtime...

Reply:
let me know if you still want to add defaults using markers ?

Reply:
> let me know if you still want to add defaults using markers ?

Yes, lets use it so its explicitly visible in the CR.
( even if it is not in use )

Reply:
updated...
I did not remove `omtiempty` though, it is harmless and is used for almost every field in the StorageCluster struct...